src/main/java/org/jenkinsci/maven/plugins/hpi/PluginWorkspaceMapImpl.java

```diff
@@ -0,0 +1,70 @@
+package org.jenkinsci.maven.plugins.hpi;
+
+import org.codehaus.plexus.component.annotations.Component;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Default and currently the only implementation of {@link PluginWorkspaceMap}
+ *
+ * @author Jesse Glick
+ * @author Kohsuke Kawaguchi
+ */
+@Component(role=PluginWorkspaceMap.class)
+public class PluginWorkspaceMapImpl implements PluginWorkspaceMap {
+    private final File mapFile;
+
+    public PluginWorkspaceMapImpl(File mapFile) {
+        this.mapFile = mapFile;
+    }
+
+    public PluginWorkspaceMapImpl() {
+        this.mapFile = new File(System.getProperty("user.home"), ".jenkins-hpl-map");
+    }
+
+    private Properties loadMap() throws IOException {
+        Properties p = new Properties();
```

Comment:
@jglick don't worry, in 2017 properties will be the new hotness file format. Trust me.